bundles/org.eclipse.swt/Eclipse SWT/gtk/org/eclipse/swt/graphics/Image.java

```diff
@@ -996,7 +996,11 @@ public Color getBackground() {
  */
 public Rectangle getBounds() {
 	if (isDisposed()) SWT.error(SWT.ERROR_GRAPHIC_DISPOSED);
-	return getBoundsInPixels();
+	if (width != -1 && height != -1) {
+		return new Rectangle(0, 0, width, height);
+	}
+	// There are no pixmaps on GTK3, so this will just return 0
+	return new Rectangle(0, 0, 0, 0);
```

Comment:
Maybe we should keep the existing method call and inline the method upon removal of the deprecated method instead of duplicating code now.